server/routes/loginAuth.js

```diff
@@ -1,47 +1,91 @@
 const router = require("express").Router();
-
-//json web token
+const passport = require("passport");
 const jwt = require("jsonwebtoken");
+const dayjs = require("dayjs");
 
-//google authentication client for token verification
-const { OAuth2Client } = require("google-auth-library");
-const client = new OAuth2Client(process.env.CLIENT_ID);
+require("./loginStrategies/googleStrategy");
 
-router.post("/google", async (req, res) => {
-    const { token } = req.body;
+router.get("/google", passport.authenticate("google", { scope: ["email", "profile"] }));
 
-    try {
-        const ticket = await client.verifyIdToken({
-            idToken: token,
-            audience: process.env.CLIENT_ID,
-        });
+// after successful login with google strategy, user db record will be sent here in the req.user property.
+router.get("/google/callback", passport.authenticate("google", { session: false }), (req, res) => {
+    // Creates a cookie with the user's login information
+    res.cookie("loginCookie", JSON.stringify(req.user), {
+        secure: false,
+        httpOnly: true,
+        expires: dayjs().add(1, "year").toDate(),
+    });
 
-        const { given_name, family_name, email, picture } = ticket.getPayload();
+    // Creates an access jwt token and stores it inside a cookie.
+    const token = jwt.sign({ id: req.user.id }, process.env.TOKEN_SECRET);
+    res.cookie("tokenCookie", token, {
+        secure: false,
+        httpOnly: true,
+    });
```

Comment:
We don't need to store the JWT inside of the cookie. We can just use the `loginCookie` to create the jwt on demand from the cookie endpoint.